otter/integration/tests/test_convergence.py

```diff
@@ -62,6 +63,130 @@ def _check_fds(_):
             return deferLater(reactor, 0, _check_fds, None)
         return self.pool.closeCachedConnections().addBoth(_check_fds)
 
+    def test_scaling_to_clb_max_after_oob_delete_type1(self):
+        """This test starts with a scaling group with no servers.  We scale up
+        to 24 servers, but after that's done, we delete 2 directly through
+        Nova.  After that, we scale up once more by 1 server, thus max'ing out
+        the CLB's ports.  We expect that the group will return to 25 servers,
+        and does not overshoot in the process.
+
+        Further, we want to make sure the deleted servers are removed from the
+        CLB.
+
+        This variant assumes a scaling group's max_entities field matches that
+        for a CLB (as of this writing).
+        """
+        return self._perform_oobd_clb_test(25)
+    test_scaling_to_clb_max_after_oob_delete_type1.timeout = 1800
+
+    def test_scaling_to_clb_max_after_oob_delete_type2(self):
+        """This test starts with a scaling group with no servers.  We scale up
+        to 24 servers, but after that's done, we delete 2 directly through
+        Nova.  After that, we scale up once more by 1 server, thus max'ing out
+        the CLB's ports.  We expect that the group will return to 25 servers,
+        and does not overshoot in the process.
+
+        Further, we want to make sure the deleted servers are removed from the
```

Comment:
Sorry, the comment about making sure deleted servers are removed from the load balancers.